scipy/stats/_stats_py.py

```diff
@@ -7422,6 +7427,10 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
         n1 = a.shape[axis]
         n2 = b.shape[axis]
 
+        if equal_var:
+            old_errstate = np.geterr()
+            np.seterr(divide='ignore', invalid='ignore')
+
```

Comment:
Is this OK for all divide and invalid errors that could occur, or just the `_var` calculations in the `trim==0` case below?

Reply:
I'm moving this inside `if trim == 0` because we have only decided that this is appropriate for the calls to `_var`.